config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -252,4 +266,27 @@ private static boolean hasAtLeastOneBeanProperty(Class<?> clazz) {
 
         return false;
     }
+
+    private static boolean isOptionalProperty(Class beanClass, PropertyDescriptor beanProp) {
+        Field field = getField(beanClass, beanProp.getName());
+        if (field == null) {
+            throw new ConfigException.BadBean("Bean property '" + beanProp + "' of class " + beanClass.getName() + " does not exist");
```

Comment:
this this exception possible? How did we get beanProp if there's no associated field?